### DISCUSSION TITLE
Filter attributes from published messages

### DISCUSSION THREAD

**Discussion by gwww** [Koenkk/zigbee2mqtt#15074] [General]
I thought `filtered_cache` can accomplish what I am looking for, but it doesn't seem to do what I thought or I have it configured wrong.

I have many, many messages published like the one below and would like to filter out attributes of the form `defaultLed1IntensityWhenOff` or more specifically `defaultLed*WhenO*`. When I set `filtered_cache` to `defaultLed1IntensityWhenOff` the attribute still appears in all the messages.

Is this what `filtered_cache` was meant for? If so, what am I doing wrong? If not, is there a way to accomplish what I want?

The problem I'm trying to solve is that Home Assistant when I receive a message such as the one below an event is generated for every one of the attributes - which results in over 100 events for one publish.

The alternative that I have considered is to used `filtered_attributes`, which I think I got to work with `defaultLed1IntensityWhenOff` but did not seem to work with `defaultLed*WhenO*`. Not sure of the regex format used but I assumed the Javascript regexes are the way to go.

The message that is being published is this:
```
MQTT publish: topic 'zigbee2mqtt/InoTest', payload '{"action":null,"activeEnergyReports":10,"activePowerReports":10,"autoTimerOff":null,"brightness":254,
"buttonDelay":"0ms","defaultLed1ColorWhenOff":null,"defaultLed1ColorWhenOn":null,"defaultLed1IntensityWhenOff":null,
"defaultLed1IntensityWhenOn":null,"defaultLed2ColorWhenOff":null,"defaultLed2ColorWhenOn":null,
"defaultLed2IntensityWhenOff":null,"defaultLed2IntensityWhenOn":null,"defaultLed3ColorWhenOff":null,
"defaultLed3ColorWhenOn":null,"defaultLed3IntensityWhenOff":null,"defaultLed3IntensityWhenOn":null,
"defaultLed4ColorWhenOff":null,"defaultLed4ColorWhenOn":null,"defaultLed4IntensityWhenOff":null,
"defaultLed4IntensityWhenOn":null,"defaultLed5ColorWhenOff":null,"defaultLed5ColorWhenOn":null,
"defaultLed5IntensityWhenOff":null,"defaultLed5IntensityWhenOn":null,"defaultLed6ColorWhenOff":null,
"defaultLed6ColorWhenOn":null,"defaultLed6IntensityWhenOff":null,"defaultLed6IntensityWhenOn":null,
"defaultLed7ColorWhenOff":null,"defaultLed7ColorWhenOn":null,"defaultLed7IntensityWhenOff":null,
"defaultLed7IntensityWhenOn":null,"defaultLevelLocal":254,"defaultLevelRemote":null,"dimmingSpeedDownLocal":null,
"dimmingSpeedDownRemote":null,"dimmingSpeedUpLocal":null,"dimmingSpeedUpRemote":null,
"doubleTapClearNotifications":null,"doubleTapUpForFullBrightness":"Button Press Event Only",
"energy":0,"firmwareUpdateInProgressIndicator":"Enabled","invertSwitch":null,"ledColorWhenOff":null,
"ledColorWhenOn":170,"ledIntensityWhenOff":null,"ledIntensityWhenOn":null,
"led_effect":{"color":136,"duration":5,"effect":"open_close","level":84},
"linkquality":102,"loadLevelIndicatorTimeout":null,"localProtection":null,"maximumLevel":null,"minimumLevel":null,
"onOffLedMode":null,"outputMode":"Dimmer","periodicPowerAndEnergyReports":3600,"power":10.1,
"powerType":"Neutral","rampRateOffToOnLocal":null,"rampRateOffToOnRemote":null,"rampRateOnToOffLocal":null,
"rampRateOnToOffRemote":null,"relayClick":"Disabled","remoteProtection":null,"smartBulbMode":null,"state":"ON",
"stateAfterPowerRestored":255,"switchType":null,"update":{"state":"idle"},"update_available":false}'
```

**Comment by gwww** [comment]
I solved this. Helps when I write a proper regEx.

**Reply by Tommixoft** [reply]
Glad you wrote the solution for others too.

**Comment by xtarke** [comment]
Is there a way to add filtered_cache parameters when a device enters the network? I already have a custom converter.

**Comment by galaxyvapers** [comment]
Thanks for addressing this! Filtering attributes from published messages will definitely help improve data accuracy and reduce noise, especially for messages related to [pod kit](https://galaxyvapers.com/products/voopoo-argus-g2-mini-pod-kit) usage. Curious if there are plans to make the filters configurable per message type for more flexibility?

**Comment by galaxyvapers** [comment]
Filtering attributes from published messages is a smart approach to streamline data processing. Just like designing [Packaging Boxes](https://thegrandpackaging.com/cosmetic-boxes/) efficiently, it ensures only relevant information is captured and reduces clutter. Implementing this can improve system efficiency and make downstream analysis more accurate.

**Comment by reddeerconcretelifting1-sys** [comment]
Red Deer Concrete Lifting is a locally trusted name for [concrete lifting Red Deer](https://reddeerconcretelifting.com/residential-void-filling/) homeowners and businesses rely on. Our polyurethane foam concrete lifting Alberta process corrects sunken slabs quickly with minimal disruption. When searching for concrete leveling near me, clients choose our professional [concrete contractors Red Deer](https://reddeerconcretelifting.com/) for long-lasting concrete repair solutions tailored to Alberta soil conditions.

**Comment by artofsmilesfontana** [comment]
Looking for a reliable [dentist in Fontana](https://artofsmilesfontana.com/)? Art of Smiles Dental Group offers family, cosmetic, and emergency dental care. Call (909)-351-4273 today.

**Comment by whereuelevate** [comment]
good but you know The job market in 2025 values experience and practical skills. Where U Elevate helps students navigate and access the [best entry-level](https://whereuelevate.com/blogs/top-8-ways-to-get-entry-level-jobs-after-graduation) opportunities efficiently.

**Comment by diamondbardentalstudio3-dotcom** [comment]
Looking for a trusted [dentist in Diamond Bar](https://diamondbardentalstudio.com/)? Diamond Bar Dental offers comprehensive, patient-focused care for families and individuals. From preventive dentistry to advanced treatments, our experienced team is committed to healthy, confident smiles with modern technology and personalized dental solutions.
